### PR TITLE
Tag filter: Fix exception when used with filter plugin

### PR DESCRIPTION
**Pull request by EricDunsworth**
When the filter and tag filter plugins were used together, typing anything into the text filter field caused the tag filter plugin to raise TypeError exceptions. Although it didn't noticeably break anything.

The exceptions were caused by some logic introduced in #9624 to fix an incompatibility between the tag filter and pagination plugins. An ``if`` condition was added to evaluate the source property of the "wb-contentupdated" event's ``data`` parameter whenever filter controls were toggled... except ``data`` doesn't always exist. Only the tag filter and data JSON plugins make use of ``data`` parameters. The filter plugin doesn't.

This fixes the issue by adjusting the ``if`` condition to ensure that the ``data`` parameter is truthy.

Also revises the edge case examples (to demonstrate that the logic fix doesn't interfere with text filter + pagination scenarios):
* Adds a text filter to example 1
* Adds a second example (based on the normal demo page's second example) with a text filter and pagination

### CODE REVIEW

**Comment by Garneauma** [comment]
Pre-approved upon successful testing.

File: src/plugins/tagfilter/tagfilter-edge-cases-fr.hbs [hunk 1]
```diff
@@ -222,20 +249,576 @@
...
+			&lt;div class="wb-tagfilter-noresult"&gt;
+				&lt;p&gt;Aucun item ne correspond à cette combinaison de filtres.&lt;/p&gt;
+			&lt;/div&gt;
+			&lt;div id="filteredEventsListPagination"&gt;&lt;/div&gt;
+		&lt;/div&gt;
+	&lt;/div&gt;
+&lt;/section&gt;</code></pre>
+</details>
+
+FR EXAMPLE 2 UPDATED:
```
Comment (Garneauma) [inline src/plugins/tagfilter/tagfilter-edge-cases-fr.hbs:270]: @EricDunsworth Is there a need to have this?

**Review by Garneauma — DISMISSED** [review]
Only one small text edit.

File: src/plugins/tagfilter/tagfilter-edge-cases-fr.hbs (hunk 1 above)
Comment (EricDunsworth) [reply src/plugins/tagfilter/tagfilter-edge-cases-fr.hbs:270]: Nope, will remove it in a moment. Btw thanks for catching it :)!

It was probably some filler content I added while prepping the new example locally... only to overlook it later on :P.

**Comment by EricDunsworth** [comment]
Just force-pushed a few minor updates:
* Removed "FR EXAMPLE 2 UPDATED:" from the new French edge case's source code
* Made the new edge case's English/French source code more consistent (EN didn't match its demo, FR wasn't truncated at the end at the same spot as EN)
  * **Side note:** Probably botched these originally since I rushed-in the new edge cases just before a technical review meeting a couple of weeks ago and lacked time for self-QA
* Updated modified dates to today

**Review by duboisp — APPROVED** [review]
Reviewed and tested.

I was able to reproduce the issue and I do confirm this fix it.

**Pull request merged**